packages/turf-dissolve/index.js

```diff
@@ -29,118 +24,54 @@ import { closestGreaterNumber } from "./lib/get-closest";
  * //addToMap
  * var addToMap = [features, dissolved]
  */
-function dissolve(featureCollection, options) {
+function dissolve(fc, options) {
   // Optional parameters
   options = options || {};
   if (!isObject(options)) throw new Error("options is invalid");
   var propertyName = options.propertyName;
 
   // Input validation
-  collectionOf(featureCollection, "Polygon", "dissolve");
+  collectionOf(fc, "Polygon", "dissolve");
 
   // Main
-  var fc = clone(featureCollection);
-  var features = fc.features;
-
-  var originalIndexOfItemsRemoved = [];
-
-  features.forEach(function (f, i) {
-    f.properties.origIndexPosition = i;
-  });
-  var tree = rbush();
-  tree.load(fc);
-
-  for (var i in features) {
-    var polygon = features[i];
-
-    var featureChanged = false;
-
-    tree.search(polygon).features.forEach(function (potentialMatchingFeature) {
-      polygon = features[i];
-
-      var matchFeaturePosition =
-        potentialMatchingFeature.properties.origIndexPosition;
-
+  var outFeatures = [];
+  if (!options.propertyName) {
+    return featureCollection([
+      multiPolygon(
```

Comment:
MultiPolygon disagrees with the @returns in the jsdoc above and the typescript types that specify polygon.

Reply:
See comments below about what we want to return